gui/WorldView.cxx

```diff
@@ -979,27 +1018,40 @@ void WorldView::computeContour(double threshold)
 //-----------------------------------------------------------------------------
 bool WorldView::loadMesh(QString const& path)
 {
+  QTE_D();
+
   QFileInfo fileInfo{path};
   if (!fileInfo.exists() || !fileInfo.isFile())
   {
     return false;
   }
 
   QGuiApplication::setOverrideCursor(Qt::WaitCursor);
-
-  // Create the vtk pipeline and read the mesh
-  vtkNew<vtkPLYReader> reader;
-  reader->SetFileName(qPrintable(path));
-  reader->Update();
-  QGuiApplication::restoreOverrideCursor();
-
-  auto mesh = vtkPolyData::SafeDownCast(reader->GetOutput());
-  if (mesh->GetPointData()->GetNumberOfArrays() <= 0)
+  QFileInfo info(path);
+  vtkSmartPointer<vtkPolyData> mesh;
+  QString ext = info.suffix().toLower();
```

Comment:
```suggestion
  vtkSmartPointer<vtkPolyData> mesh;
  auto const& ext = fileInfo.suffix().toLower();
```

Reply:
I am not a big fan of auto in these simpler cases where it does not even save space but make the type less clear. You have to lookup the return of toLower to see that the type is. But I will use the project const placement.

Reply:
Using `const` would be good. The "biggest" change here though is not gratuitously recreating the `QFileInfo`.

Reply:
Sounds good.